Warehouse.Infrastructure/RestClient/Products/ProductsRestClient.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Warehouse.Domain.Entities;
+using Warehouse.Infrastructure.RestClient.Configuration;
+
+namespace Warehouse.Infrastructure.RestClient.Products
+{
+	public class ProductsRestClient
+	{
+        private readonly RestClient<Product, string> _restClient;
+
+        public ProductsRestClient(MockyProductsConfiguration configuration)
+        {
+            _restClient = new RestClient<Product, string>(configuration.Url);
```

Comment:
You are coupling your ProductsRestClient with the new RestClient.
The [Dependency inversion principle](https://en.wikipedia.org/wiki/Dependency_inversion_principle): "Depend upon abstractions, [not] concretes."

When designing object-oriented systems, it is important to follow the Dependency Inversion Principle (DIP), which is part of the SOLID principles. The main idea of DIP is to use abstractions instead of concrete implementations. By separating high-level modules with complex features from low-level modules that provide basic operations, DIP promotes the use of an abstraction layer to reduce coupling between the modules.When designing object-oriented systems, it is important to follow the Dependency Inversion Principle (DIP), which is part of the SOLID principles. The main idea of DIP is to use abstractions instead of concrete implementations. By separating high-level modules with complex features from low-level modules that provide basic operations, DIP promotes the use of an abstraction layer to reduce coupling between the modules.
https://dotnettutorials.net/lesson/real-time-examples-of-dependency-inversion-principle-in-csharp/

You have to think how to implement it in a different way, otherwise it looks like an interesting approach.

Reply:
Done.